py/asmrv32.c

```diff
@@ -270,6 +270,15 @@ static void emit_function_epilogue(asm_rv32_t *state, mp_uint_t registers) {
     state->saved_registers_mask = old_saved_registers_mask;
 }
 
+static void calculate_displacement_for_label(asm_rv32_t *state, mp_uint_t label, ptrdiff_t *displacement, bool *emit_short_jump) {
```

Comment:
I suggest making this return `bool` for the `emit_short_jump` variable, instead of taking that as a pointer-to-return.  That makes the code a little neater, and maybe more efficient because it's not taking the address of a local variable.

Also, I suggest calling the variable `can_emit_short_jump` because it's not a requirement to emit a short jump, just a hint that it's possible.

Reply:
The function not returning a `bool` was deliberate, as it's more difficult to forget about the second returned value that way, since it is forced to be declared for the code to compile (and the compiler would reorder everything anyway when optimising things - there are *a lot* of unused registers on RV32 after all).  I agree with the variable renaming, it is definitely clearer that way!

That said, I've implemented your suggestions, thanks for taking a look.